test/models/accommodation_test.rb

```diff
@@ -0,0 +1,96 @@
+    assert_not build(:accommodation, percent_time_expansion: -1).valid?
+    assert_not build(:accommodation, percent_time_expansion: -1.5).valid?
+    assert_not build(:accommodation, percent_time_expansion: 0.2).valid?
+    assert_not build(:accommodation, percent_time_expansion: 1.5).valid?
+
+    assert build(:accommodation, percent_time_expansion: 0).valid?
+    assert build(:accommodation, percent_time_expansion: 1).valid?
+    assert build(:accommodation, percent_time_expansion: 50).valid?
+    assert build(:accommodation, percent_time_expansion: 100).valid?
+    assert build(:accommodation, percent_time_expansion: 125).valid?
+  end
+
+  test 'factor calculation' do
+    acc = build(:accommodation, percent_time_expansion: 25)
+    assert_equal acc.factor, 1.25
+
+    acc.percent_time_expansion = 0
+    assert_equal acc.factor, 1
+
+    acc.percent_time_expansion = 100
+    assert_equal acc.factor, 2
+  end
+
+  test 'accommodated duration with expansion' do
+    acc = build(:accommodation, percent_time_expansion: 25)
+    reg = acc.registration
+    exam = reg.exam
+    extra_duration = exam.duration / 4.0
+    assert_equal extra_duration, reg.accommodated_extra_duration
+    assert_equal exam.duration + extra_duration, reg.accommodated_duration
+  end
+
+  test 'accommodated end time with expansion' do
+    acc = build(:accommodation, percent_time_expansion: 25)
+    reg = acc.registration
+    exam = reg.exam
+    extra_duration = exam.duration / 4.0
+    assert_equal exam.end_time + extra_duration, reg.accommodated_end_time
+  end
+
+  test 'start time is updated to accommodated start time' do
+    reg = build(:registration)
+    exam = reg.exam
+    new_start_time = exam.start_time - 1.hour
+    acc = build(:accommodation, registration: reg, new_start_time: new_start_time)
+    assert_equal new_start_time.to_i, reg.accommodated_start_time.to_i
+  end
+
+  test 'end time is adjusted by factor' do
+    acc = build(:accommodation, percent_time_expansion: 25)
+    reg = acc.registration
+    exam = reg.exam
+    extra_duration = exam.duration / 4.0
+    assert_equal exam.start_time.to_i, reg.accommodated_start_time.to_i
+    assert_equal (exam.end_time + extra_duration).to_i, reg.accommodated_end_time.to_i
+  end
+
+  test 'full time remaining for an early start' do
+    reg = build(:registration, :early_start)
+    exam = reg.exam
+    acc = build(:accommodation, registration: reg, percent_time_expansion: 25)
+    extra_duration = exam.duration / 4.0
+    assert_equal exam.duration + extra_duration, reg.accommodated_duration
+    assert_equal reg.accommodated_duration, reg.effective_duration
+  end
+
+  test 'more time remaining for a late start with accommodation' do
+    reg = build(:registration, :late_start)
+    exam = reg.exam
+    acc = build(:accommodation, registration: reg, percent_time_expansion: 50)
+    acc.save
+
+    # # The student has a 50% larger window, and 50% more time.
```

Comment:
I think this comment is misleading -- they don't have a 50% larger window; they just have the 50% extra time added to their normal window, right?